main.go

```diff
@@ -1,147 +1,187 @@
 package main
 
 import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
 	"os"
+	"regexp"
 	"strconv"
-
-	"bufio"
 	"strings"
 
-	"regexp"
-
-	"fmt"
-
 	"github.com/bitrise-io/go-utils/command"
 	"github.com/bitrise-io/go-utils/fileutil"
 	"github.com/bitrise-io/go-utils/log"
 	"github.com/bitrise-tools/go-steputils/input"
 )
 
-func logFail(format string, v ...interface{}) {
-	log.Errorf(format, v...)
-	os.Exit(1)
+// ConfigsModel ...
+type ConfigsModel struct {
+	BuildGradlePth    string
+	NewVersionName    string
+	NewVersionCode    string
+	VersionCodeOffset string
+}
+
+func createConfigsModelFromEnvs() ConfigsModel {
+	return ConfigsModel{
+		BuildGradlePth:    os.Getenv("build_gradle_path"),
+		NewVersionName:    os.Getenv("new_version_name"),
+		NewVersionCode:    os.Getenv("new_version_code"),
+		VersionCodeOffset: os.Getenv("version_code_offset"),
+	}
+}
+
+func (configs ConfigsModel) print() {
+	log.Infof("Configs:")
+	log.Printf("- BuildGradlePth: %s", configs.BuildGradlePth)
+	log.Printf("- NewVersionName: %s", configs.NewVersionName)
+	log.Printf("- NewVersionCode: %s", configs.NewVersionCode)
+	log.Printf("- VersionCodeOffset: %s", configs.VersionCodeOffset)
+}
+
+func (configs ConfigsModel) validate() error {
+	if err := input.ValidateIfPathExists(configs.BuildGradlePth); err != nil {
+		return errors.New("issue with input BuildGradlePth: " + err.Error())
+	}
+	return nil
+}
+
+type updateFn func(line string, lineNum int, matches []string) string
+
+func findAndUpdate(reader io.Reader, update map[string]updateFn) (string, error) {
+	scanner := bufio.NewScanner(reader)
+	updatedLines := []string{}
+
+	reByPattern := make(map[string]*regexp.Regexp, len(update))
```

Comment:
Why do you make a map for the compiled regexps? Just use the `regexp.MustCompile(patter)` in line _66_.